pkgs/applications/editors/emacs/make-emacs.nix

```diff
@@ -185,20 +185,20 @@ mkDerivation (finalAttrs: {
     # Reduce closure size by cleaning the environment of the emacs dumper
     ''
       substituteInPlace src/Makefile.in \
-        --replace 'RUN_TEMACS = ./temacs' 'RUN_TEMACS = env -i ./temacs'
+        --replace-fail 'RUN_TEMACS = ./temacs' 'RUN_TEMACS = env -i ./temacs'
     ''
 
     ''
       substituteInPlace lisp/international/mule-cmds.el \
-        --replace /usr/share/locale ${gettext}/share/locale
-
+        --replace-fail /usr/share/locale ${gettext}/share/locale
+    ''
+  ] ++ lib.optionals (lib.versionOlder version "29") [
+    ''
       for makefile_in in $(find . -name Makefile.in -print); do
-        substituteInPlace $makefile_in --replace /bin/pwd pwd
+        substituteInPlace $makefile_in --replace-fail /bin/pwd pwd
```

Comment:
This makes Emacs 28 (and maybe more) fail to build because not all founded makefiles have /bin/pwd.  You can run `find . -name Makefile.in -print0 | xargs --verbose -0 -I '{}' -n 1 rg /bin/pwd '{}'` in the Emacs 28.2 source code repo to verify this.

Reply:
Solved.